scripts/suite_objects.py

```diff
@@ -1195,7 +1203,238 @@ def analyze(self, phase, group, scheme_library, suite_vars, level):
+        if dvar:
+            var_in_call_list = True
+        else:
+            var_in_call_list = False
+            for var_dict in self.__group.suite_dicts():
+                dvar = var_dict.find_variable(standard_name=standard_name)
+                if dvar:
+                    break
+        if not dvar:
+            raise Exception(f"No variable with standard name '{standard_name}' in var_dicts")
+        local_name = dvar.get_prop_value('local_name')
+
+        # If the variable is allocatable or a pointer and the intent for the
+        # scheme is 'out', then we can't test anything because the scheme is
+        # going to allocate the variable or associate the pointer. We don't have
+        # this information earlier in add_var_debug_check, therefore need to back
+        # out here, using the information from the scheme variable (call list).
+        svar = self.call_list.find_variable(standard_name=standard_name)
+        intent = svar.get_prop_value('intent')
+        if intent == 'out' and (allocatable or pointer):
+            return
+
+        # Get the condition on which the variable is active
+        (conditional, _) = var.conditional(var_dicts)
+
+        # For scalars, assign to dummy variable if the variable intent is in/inout
+        if not dimensions:
+            if not intent == 'out':
+                dummy_lname = dummy.get_prop_value('local_name')
+                outfile.write(f"if ({conditional}) then", indent)
+                outfile.write(f"! Assign value of {local_name} to dummy", indent+1)
+                outfile.write(f"{dummy_lname} = {local_name}", indent+1)
+                outfile.write(f"end if", indent)
+        # For arrays, check size of array against dimensions in metadata, then assign
+        # the lower and upper bounds to the dummy variable if the intent is in/inout
+        else:
+            array_size = 1
+            dim_strings = []
+            lbound_strings = []
+            ubound_strings = []
+            for dim in dimensions:
+                if not ':' in dim:
+                    # In capgen, any true dimension (that is not a single index) does
+                    # have a colon (:) in the dimension, therefore this is an index
+                    for var_dict in var_dicts:
+                        dvar = var_dict.find_variable(standard_name=dim, any_scope=False)
+                        if dvar is not None:
+                            break
+                    if not dvar:
+                        raise Exception(f"No variable with standard name '{dim}' in var_dicts")
+                    dim_lname = dvar.get_prop_value('local_name')
+                    dim_length = 1
+                    dim_strings.append(dim_lname)
+                    lbound_strings.append(dim_lname)
+                    ubound_strings.append(dim_lname)
+                else:
+                    # I don't know how to do this better. Schemes can rely on the host cap
```

Comment:
This is where I would love to get some help. I think that there is logic that matches horizontal dimensions that can clean up this part by a lot.

Reply:
See https://github.com/climbfuji/ccpp-framework/pull/2

Reply:
I am pretty sure that that's what I had first, and it didn't work because of the "magic" that happens when translating horizontal dimensions. I'll give it a try tough, thanks.

Reply:
This works, but one difference. ncol is not used, instead col_end-start+1 is, which is ncol.

Reply:
That's neat. I am worried about the cap's module variables, though, because these use different indices when passed to the physics (`cols:cole`). But maybe I am wrong. I need to try this over the weekend or next week, and hopefully it works for that, too. Thanks for your help @dustinswales !

Reply:
@dustinswales None of  that logic seemed appropriate and sufficient, therefore I went ahead and did what we discussed at the last tag up: pull out the logic that handles the horizontal dimension from `write_var_debug_check` and put it in its own routine.